File: server/src/models/game/dice.ts

export interface Dice {
    getName: () => string,
    getMax: () => number,
    getValue: () => number | null,
    roll: () => number,
}

export class D6 implements Dice {
    private name: string;
    private max: number;
    private value: number | null;

    constructor() {
        this.name = "D6";
        this.max = 6;
        this.value = null;
    }

    getName() {
        return this.name;
    }

    getMax() {
        return this.max;
    }

    getValue() {
        return this.value;
    }

    roll(): number {
        this.value = Math.floor(Math.random() * this.max) + 1; 
        return this.value;
    }
}

File: app/src/main/java/com/example/liars_dice/model/game/Dice.java
package com.example.liars_dice.model.game;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Dice {
    private String name;
    private Integer max;
    private Integer value;

    public Dice(String name, Integer max) {
        this.name = name;
        this.max = max;
        this.value = null;
    }

    public Dice(
            @JsonProperty("name") String name,
            @JsonProperty("max") Integer max,
            @JsonProperty("value") Integer value
    ) {
        this.name = name;
        this.max = max;
        this.value = value;
    }

    public String getName() {
        return name;
    }

    public Integer getMax() {
        return max;
    }

    public Integer getValue() {
        return value;
    }
}
